org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -2199,13 +2191,19 @@ protected void setUpGeneralParameters() {
             targetConfig.compileDefinitions.put("MODAL_REACTORS", "TRUE");
         }
         if (targetConfig.threading && targetConfig.platformOptions.platform == Platform.ARDUINO) {
-
             //Add error message when user attempts to set threading=true for Arduino
             if (targetConfig.setByUser.contains(TargetProperty.THREADING)) {
                 errorReporter.reportWarning("Threading is incompatible on Arduino. Setting threading to false.");
             }
             targetConfig.threading = false;
         }
+        if (!targetConfig.noCompile && targetConfig.platformOptions.platform == Platform.ARDUINO) {
```

Comment:
Why only do this if `!targetConfig.noCompile`?

Reply:
I'm trying to send a warning saying that the LFC won't compile for the Arduino Platform and that you must use Arduino-CLI to complete compilation.

Reply:
I guess I was thrown off by the comment on line `2201`, which mentions `threading=true` which seems to have nothing to do with this.